File: E-class/e-commerce-backend/src/main/java/com/vn/backend/service/AttributeValueService.java
package com.vn.backend.service;


import com.vn.backend.dto.request.AttributeValueRequest;
import com.vn.backend.dto.response.AttributeValueResponse;

import java.util.List;

public interface AttributeValueService {
    List<AttributeValueResponse> getByCode(String code);
    AttributeValueResponse createByCode(String code, AttributeValueRequest req);
    AttributeValueResponse update(Long id, AttributeValueRequest req);
    void disable(Long id);
}

File: E-class/e-commerce-frontend/src/services/attributeValue.service.ts
import { axiosClient } from "./axiosClient";
import type { AttributeValue, AttributeValueRequest } from "@/features/attribute/attributeValue.model";

export const attributeValueService = {
  getByCode: async (code: string): Promise<AttributeValue[]> => {
    const res = await axiosClient.get<AttributeValue[]>(`/v1/attributes/${code}/values`);
    return res.data;
  },

  createByCode: async (code: string, payload: AttributeValueRequest): Promise<AttributeValue> => {
    const res = await axiosClient.post<AttributeValue>(`/v1/attributes/${code}/values`, payload);
    return res.data;
  },

  update: async (id: number, payload: AttributeValueRequest): Promise<AttributeValue> => {
    const res = await axiosClient.put<AttributeValue>(`/v1/attribute-values/${id}`, payload);
    return res.data;
  },

  disable: async (id: number): Promise<void> => {
    await axiosClient.delete(`/v1/attribute-values/${id}`);
  },
};